### DISCUSSION TITLE
A guide to installing spacepy on windows 10?

### DISCUSSION THREAD

**Discussion by nsioulas** [spacepy/spacepy#470] [Support and Q&A] [answered]
Thanks!

**Comment by drsteve** [comment, marked as answer]
I'm not sure what problems (if any) you've hit along the way, but we do have an installation guide in the [documentation](https://spacepy.github.io/).
Basic installation documentation is at https://spacepy.github.io/install.html, with some windows-specific guidance at https://spacepy.github.io/install_windows.html

The short answer is that we recommend installing [anaconda](https://docs.anaconda.com/anaconda/install/windows/).
Once you have that installed, fire up an anaconda command window (should be in your start menu after installation).
We don't specifically talk about virtual environments in our installation guide, but it's good practice. So assuming you've got a fresh installation of anaconda, then your one-time setup will look like:
```
conda create -n default python=3.8 numpy
```
Here `default` is just the name for the environment - call it whatever you want. You can also specify whatever version of python you want. 3.9 is still lagging in support from some other packages, so I'd use 3.8 right now, but we support 3.9 (and 3.6, 3.7). The `numpy` on the end tells conda to also install numpy in the new environment. Spacepy needs that to install.
If you already have a virtual environment that you use, then skip this step.

So now you have conda and a virtual environment with numpy. Activate your environment, then install spacepy:
```
conda activate default
pip install spacepy
```
That's it.
There will be a couple of bits of spacepy that won't work without extra packages being installed (see [the windows instructions](https://spacepy.github.io/install_windows.html)), but that should get you going.

The most likely cause of issues is, in my experience, having things set up for a system Python install as well as having Anaconda installed. That can (depending on how things got set up) lead to python trying to find incorrect versions of things. Another likely issue is that the very latest version of numpy (1.19.4) doesn't work on windows. So you'll need 1.19.3 or earlier. 


If you don't want to use anaconda for your Python on Windows, then you have two options.
- The first is more powerful, but a bit fiddly, which is using WSL (Windows Subsystem for Linux) - that basically runs Linux on your Windows machine. I've been really impressed with the new WSL2, and how it integrates with the Visual Studio Code editor. My preferred terminal is [MobaXTerm](https://mobaxterm.mobatek.net/), which also bundles an X11 server so you can just launch a WSL terminal and anything that opens graphical windows will work as expected. Getting WSL2 set up can take a little work, followed by setting up a new bare-bones linux environment, hence we don't provide instructions.
- The second is to just use a standard [Windows Python install](https://www.python.org/downloads/windows/). Once you have that you can install numpy using pip, and then `pip install spacepy` should grab the latest release and install it (same caveats as noted above), Although note that [numpy also recommend](https://numpy.org/install/) using conda.

(Just for clarity, Anaconda is a full set of packages and tools, one of which is the `conda` package manager. Conda is what you need here, but the full anaconda installation does make life a bit easier.)


Finally, what if you have issues? Well, the easiest way for us to help is if we can see exactly what problems you were having. If there are error messages, please copy & paste as much as possible into a comment here. If it's actually a problem on our end then we can open an issue to get it fixed. If it's not a problem on our end then we'll try to help diagnose what the issue and and how to work around it here in discussion.

**Reply by nsioulas** [reply]
Thank you very much for the detailed response!

**Reply by jtniehof** [reply]
Closing now as out of date.
